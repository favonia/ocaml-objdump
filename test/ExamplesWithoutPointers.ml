let test x = Format.printf "%a@." Objdump.pp x

type t = A | B of int | C

type _ u = ..
type _ u += D : int u
type _ u += E : 'a -> 'a u

type v = ..
type v += U : 'a u -> v

let () = test ()
let () = test A
let () = test (B 100)
let () = test C
let () = test 42
let () = test false
let () = test (-1)
let () = test None
let () = test (Some (Some (Some (Some (Some (Some (Some (Some 10))))))))
let () = test (Failure "str")
let () = test (Invalid_argument "str")
let () = test "a very very very very very very very long string with \x012 and \" and ' and \\"
let () = test (("fst", "snd"), 2, 3.0, ())
let () = test 3.14
let () = test (2.18, "a")
let () = test (U D)
let () = test (U (E 10))
let () = let m = lazy (1 + 2) in let _ = Lazy.force m in test m
let () = let rec m = lazy (test m) in Lazy.force m
let () = test [|1; 2; 3|]
let () = test [|1.0; 2.0; 3.0|]
let () = test [1; 2; 3]
let () = test Float.nan
let () = test 10l
let () = test 20L
let () = test 30n
