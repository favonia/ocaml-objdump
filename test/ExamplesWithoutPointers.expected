int(0)
int(0)
variant0(int(100))
int(1)
int(42)
int(0)
int(-1)
int(0)
variant0(
  variant0(
    variant0(variant0(variant0(variant0(variant0(variant0(int(10)))))))))
variant0(object{class=string("Failure"),object-id=int(-3)},string("str"))
variant0(
  object{class=string("Invalid_argument"),object-id=int(-4)},
  string("str"))
string(
  "a very very very very very very very long string with \0012 and \" and ' and \\")
variant0(
  variant0(string("fst"),string("snd")),
  int(2),
  double(3.000000),
  int(0))
double(3.140000)
variant0(double(2.180000),string("a"))
variant0(
  object{
    class=string("Dune__exe__ExamplesWithoutPointers.U"),
    object-id=int(11)},
  object{
    class=string("Dune__exe__ExamplesWithoutPointers.D"),
    object-id=int(9)})
variant0(
  object{
    class=string("Dune__exe__ExamplesWithoutPointers.U"),
    object-id=int(11)},
  variant0(
    object{
      class=string("Dune__exe__ExamplesWithoutPointers.E"),
      object-id=int(10)},
    int(10)))
forward(int(3))
forcing(<0x1>)
variant0(int(1),int(2),int(3))
double_array(1.,2.,3.)
variant0(int(1),variant0(int(2),variant0(int(3),int(0))))
